### PR TITLE
Enable docstring checker and fix comments

### PR DESCRIPTION
**Pull request by typhoonzero**
Enable docstring checker to enhance document in python codes.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -2959,14 +3049,32 @@ def sequence_reshape(input, new_dim):
     return out
 
 
-@autodoc()
+# FIXME(wuyi): let docstring_checker.py under stand @autodoc.
```
Comment (gongweibao) [inline python/paddle/fluid/layers/nn.py]: understand

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -3024,8 +3132,9 @@ def transpose(x, perm, name=None):
     perm[i]-th dimension of `input`.
 
     Args:
-       input (Variable): (Tensor), A Tensor.
-       perm (list): A permutation of the dimensions of `input`.
+        x (Variable): (Tensor), A Tensor.
```
Comment (gongweibao) [inline python/paddle/fluid/layers/nn.py]: Variable or Tensor?

File: tools/codestyle/docstring_checker.py [hunk 3]
```diff
@@ -320,15 +327,19 @@ def all_args_in_doc(self, node, doc):
...
+                node=node,
+                line=node.fromlineno,
+                args=list(args_not_documented))
             return False
 
         for t in args:
             if t not in parsed_args:
-                print t, " with (type) not in ", parsed_args
-                self.add_message('W9003', node=node, line=node.fromlineno)
+                self.add_message(
```
Comment (gongweibao) [inline tools/codestyle/docstring_checker.py]: Thanks!

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (typhoonzero) [reply python/paddle/fluid/layers/nn.py]: Done.

**Pull request merged**